### PR TITLE
added type hints for backtracking/minimax.py

### PR DESCRIPTION
**Pull request by Abdujabbor**
Added type hints and little modification for readability on backtracking/minimax.py

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by Abdujabbor** [comment]
@cclauss  look I think it's better than the previous version

**Review by shellhub — APPROVED** [review]

File: backtracking/minimax.py [hunk 1]
```diff
@@ -9,26 +9,54 @@
...
-        minimax(Depth + 1, nodeIndex * 2 + 1, True, scores, height),
+        minimax(depth + 1, node_index * 2, True, scores, height),
+        minimax(depth + 1, node_index * 2 + 1, True, scores, height),
     )
 
 
 if __name__ == "__main__":
+    import doctest
 
-    scores = [90, 23, 6, 33, 21, 65, 123, 34423]
```
Comment (cclauss) [inline backtracking/minimax.py:30]: These five lines show the user something when they run the program.  Removing them means that the program shows the user nothing when run.

Comment (Abdujabbor) [reply backtracking/minimax.py:30]: @cclauss thank you for your feedback, I have added these lines as method main, look at it if is it right way or not?

File: backtracking/minimax.py [hunk 2]
```diff
@@ -1,4 +1,5 @@
 from typing import List
```
Comment (cclauss) [inline backtracking/minimax.py:1]: Please rebase with master to pick up the #2464 changes.

**Comment by Abdujabbor** [comment]
@cclauss I can't understand why but after rebase there are a lot of other commits in my pull request, is it ok? or better to close this PR and send another with only my changes?

**Comment by cclauss** [comment]
Please create a new PR with just one file changed.

**Pull request closed without merging**